include/aikido/util/yaml_utils.hpp

```diff
@@ -0,0 +1,202 @@
+#include <yaml-cpp/yaml.h>
+
+
+namespace aikido {
+namespace util {
+
+template <class _Scalar,
+          int _Rows, int _Cols, int _Options, int _MaxRows, int _MaxCols>
+inline void deserialize(
+    YAML::Node const &node,
+    Eigen::Matrix<_Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols> &matrix)
+{
+    typedef Eigen::Matrix<_Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols> MatrixType;
+    typedef typename MatrixType::Index Index;
+    typedef typename MatrixType::Scalar Scalar;
+
+    using boost::format;
+    using boost::str;
+    using std::runtime_error;
+
+    if (node.Type() != YAML::NodeType::Sequence) {
+        throw runtime_error("Matrix or vector must be a sequence.");
+    }
+
+    Index const rows = node.size();
+    if (MatrixType::RowsAtCompileTime != Eigen::Dynamic 
+     && rows != MatrixType::RowsAtCompileTime) {
+        throw runtime_error(str(
+            format("Matrix has incorrect number of rows: expected %d; got %d.")
+            % MatrixType::RowsAtCompileTime % rows));
+    }
+
+    if (node.Tag() == "Vector" || node.Tag() == "!Vector") {
+        matrix.resize(rows, 1);
+        for (Index i = 0; i < rows; ++i) {
+            matrix(i, 0) = node[i].template as< _Scalar >();
+        }
+    } else if (node.Tag() == "Matrix" || node.Tag() == "!Matrix") {
+        Index const cols = node[0].size();
+
+        if (MatrixType::ColsAtCompileTime != Eigen::Dynamic 
+         && cols != MatrixType::ColsAtCompileTime) {
+            throw runtime_error(str(
+                format("Matrix has incorrect number of cols: expected %d; got %d.")
+                % MatrixType::ColsAtCompileTime % cols));
+        }
+
+        matrix.resize(rows, cols);
+
+        for (Index r = 0; r < node.size(); ++r) {
+            if (node[r].Type() != YAML::NodeType::Sequence) {
+                throw runtime_error(str(
+                    format("Row %d of the matrix must be a sequence.") % r));
+            } else if (node[r].size() != cols) {
+                throw runtime_error(boost::str(
+                    format("Expected row %d to have %d columns; got %d.")
+                        % r % cols % node[r].size()));
+            }
+
+            for (Index c = 0; c < cols; ++c) {
+                matrix(r, c) = node[r][c].template as<_Scalar>();
+            }
+        }
+    } else {
+        throw runtime_error(str(
+            format("Unknown type of matrix '%s'.") % node.Tag()));
+    }
+}
+
+template <class _Scalar, int Dim, int Mode, int _Options>
+inline void deserialize(YAML::Node const &node,
+                        Eigen::Transform<_Scalar, Dim, Mode, _Options> &pose)
+{
+    deserialize(node, pose.matrix());
+}
+
+template <class T>
+inline bool has_child(YAML::Node const &node, T const &key)
+{
+	return node[key].IsDefined();
+}
+
+} // namespace util
+} // namespace aikido
+
+
+namespace YAML {
+namespace detail {
+
+template <typename MatrixType, bool IsVectorAtCompileTime>
+struct encode_impl {
```

Comment:
This is an internal helper class, so it deserves a docstring. 😁

The reason this exists is so we can use template specialization to switch between serializing vectors and serializing matrices based on the `IsVectorAtCompileTime` flag. This is a "nice to have" that lets us generate `!Vector [1, 2, 3]` instead of the syntactic travesty like `!Matrix [[1], [2] ,[3]]` when serializing an Eigen vector.